pkg/operator/ceph/cluster/osd/osd.go

```diff
@@ -600,14 +600,17 @@ func (c *Cluster) startOSDDaemonsOnPVC(pvcName string, config *provisionConfig,
 				logger.Infof("deployment for osd %d already exists. updating if needed", osd.ID)
 				if err = updateDeploymentAndWait(c.context, c.clusterInfo, dp, opconfig.OsdType, strconv.Itoa(osd.ID), c.spec.SkipUpgradeChecks, c.spec.ContinueUpgradeAfterChecksEvenIfNotHealthy); err != nil {
 					logger.Errorf("failed to update osd deployment %d. %v", osd.ID, err)
+					// TODO: should orch status be updated here also?
 				}
 			} else {
 				// we failed to create job, update the orchestration status for this pvc
 				logger.Warningf("failed to create osd deployment for pvc %q, osd %v. %v", osdProps.pvc.ClaimName, osd, createErr)
+				// TODO: orch status is not updated for this PVC as stated above
 				continue
 			}
 		}
 
+		// TODO: why are we doing this a second time? it already happens above
 		if createErr != nil && kerrors.IsAlreadyExists(createErr) {
 			if err = updateDeploymentAndWait(c.context, c.clusterInfo, dp, opconfig.OsdType, strconv.Itoa(osd.ID), c.spec.SkipUpgradeChecks, c.spec.ContinueUpgradeAfterChecksEvenIfNotHealthy); err != nil {
 				logger.Errorf("failed to update osd deployment %d. %v", osd.ID, err)
```

Comment:
@leseb right now I am still investigating the code to determine the minimal set of changes I can make to get the behavior we want to create new OSDs before updating existing ones. I have some questions marked by `// TODO` comments that I'm hoping you can answer for me.